api_v3/lib/types/entry/KalturaPlayableEntry.php

```diff
@@ -106,4 +106,18 @@ public function toObject($dbObject = null, $skip = array())
 			
 		return $dbObject;
 	}
+
+	public function doFromObject($sourceObject, KalturaDetachedResponseProfile $responseProfile = null)
+	{
+		parent::doFromObject($sourceObject, $responseProfile);
+		$recordedLengthInMs = $sourceObject->getRecordedLengthInMsecs();
+		if ($sourceObject->getIsRecordedEntry() && $recordedLengthInMs > 0)
```

Comment:
I think this condition should also include && if ! !$sourceObject->getLengthInMsecs()
I see that every time setLengthInMsecs is called this field value is reseted to 0 and once the entry replacement will happen and the entry will get it duration wont the API return duration of 0 in the response?